hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/ingest/IngestChecker.java

```diff
@@ -15,62 +15,131 @@
+public class IngestChecker {
+
+    private static final Logger LOG = LoggerFactory.getLogger(IngestChecker.class);
+
+    private final AccountID nodeAccountID;
+    private final TransitionLogicLookup transitionLogic;
+    private final GlobalDynamicProperties dynamicProperties;
 
     /**
-     * Validates a {@link Transaction}
+     * Constructor of the {@code IngestChecker}
      *
-     * @param tx the {@code Transaction}
-     * @throws PreCheckException if validation fails
-     * @throws NullPointerException if {@code tx} is {@code null}
+     * @param nodeAccountID the {@link AccountID} of the <em>node</em>
+     * @param transitionLogic a {@link TransitionLogicLookup}
+     * @param dynamicProperties the {@link GlobalDynamicProperties}
+     * @throws NullPointerException if one of the arguments is {@code null}
      */
-    void checkTransaction(@NonNull Transaction tx) throws PreCheckException;
+    public IngestChecker(
+            @NonNull final AccountID nodeAccountID,
+            @NonNull final TransitionLogicLookup transitionLogic,
+            @NonNull final GlobalDynamicProperties dynamicProperties) {
+        this.nodeAccountID = nodeAccountID;
+        this.transitionLogic = requireNonNull(transitionLogic);
+        this.dynamicProperties = requireNonNull(dynamicProperties);
+    }
 
     /**
-     * Validates a {@link SignedTransaction}
+     * Checks a transaction for semantic errors
      *
-     * @param tx the {@code SignedTransaction} to check
-     * @throws PreCheckException if validation fails
-     * @throws NullPointerException if {@code tx} is {@code null}
+     * @param txBody the {@link TransactionBody}
+     * @param functionality the {@link HederaFunctionality} of the transaction
+     * @throws NullPointerException if one of the arguments is {@code null}
+     * @throws PreCheckException if a semantic error was discovered. The contained {@code
+     *     responseCode} provides the error reason.
      */
-    void checkSignedTransaction(@NonNull SignedTransaction tx) throws PreCheckException;
+    public void checkTransactionSemantics(
+            @NonNull final TransactionBody txBody, @NonNull final HederaFunctionality functionality)
+            throws PreCheckException {
+        final ResponseCodeEnum errorCode;
+
+        var txnId = txBody.getTransactionID();
+        if (!Objects.equals(nodeAccountID, txBody.getNodeAccountID())) {
+            throw new PreCheckException(INVALID_NODE_ACCOUNT);
+        }
+
+        if (txnId.getScheduled() || txnId.getNonce() != USER_TRANSACTION_NONCE) {
+            throw new PreCheckException(TRANSACTION_ID_FIELD_NOT_ALLOWED);
+        }
+
+        if (functionality == TokenAccountWipe) {
+            errorCode =
+                    TokenWipeAccessor.validateSyntax(
+                            txBody.getTokenWipe(),
+                            dynamicProperties.areNftsEnabled(),
+                            dynamicProperties.maxBatchSizeWipe());
```

Comment:
This looks like it is in the wrong place. In the diagram, "Check Semantics" is a green box, meaning it is implemented by the service modules. Probably it is a shared responsibility, where common transaction semantics are checked here, but service-specific semantics are delegated to the service. Maybe a new "handler" is required on Services for this? "Precheck Transaction Handler" or something?

Reply:
Yes, the special handling of TokenAccountWipe is weird. This is just a temporary solution. @Neeharika-Sompalli mentioned they are working on a redesign. As soon as the new solution is available, I will fix this section.

Reply:
@netopyr That redesign is stopped, as we did not pursue extending alias support. But, we can remove this once the precheck validations for each transaction are implemented.

Reply:
Removed it, it became obsolete with the new handler design.